Code/src/main/java/nl/utwente/group10/ui/components/blocks/FunctionBlock.java

```diff
@@ -169,14 +202,19 @@ public final StringProperty typeProperty() {
         return type;
     }
 
+    /** Returns the IntegerPorperty for the bowtie index of this FunctionBlock. */
+    public final IntegerProperty bowtieIndexProperty() {
+        return bowtieIndex;
+    }
+
     @Override
     public boolean inputsAreConnected() {
-        return inputs.stream().allMatch(ConnectionAnchor::isConnected);
+        return getActiveInputs().stream().allMatch(ConnectionAnchor::isConnected);
     }
 
     @Override
     public boolean inputIsConnected(int index) {
-        return index>=0 && index < inputs.size() && inputs.get(index).isConnected();
+        return index >= 0 && index < getBowtieIndex() && inputs.get(index).isConnected();
```

Comment:
Handmatige bounds check is wat mij betreft niet nodig. `List.get` gooit een IndexOutOfBoundsException als de index buiten het bereik ligt, en dat is denk ik precies wat je hier wilt doen.